Formula/alp.rb

```diff
@@ -0,0 +1,28 @@
+class Alp < Formula
+  desc "Access Log Profiler"
+  homepage "https://github.com/tkuchiki/alp"
+  url "https://github.com/tkuchiki/alp/archive/v1.0.1.tar.gz"
+  sha256 "7206f60efcca9a4f149d6abdcaf806f290fcf16237b1f7528e40628c37816449"
+  head "https://github.com/tkuchiki/alp.git"
+
+  depends_on "dep" => :build
+  depends_on "go" => :build
+
+  def install
+    ENV["GOPATH"] = buildpath
+    ENV["GLIDE_HOME"] = HOMEBREW_CACHE/"glide_home/#{name}"
+
+    srcpath = buildpath/"src/github.com/tkuchiki/alp"
+    srcpath.install buildpath.children
+
+    cd srcpath do
+      system "dep", "ensure", "-vendor-only"
+      system "go", "build", "-o", bin/"alp", "cli/alp/main.go"
+      prefix.install_metafiles
+    end
+  end
+
+  test do
+    system "#{bin}/alp", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
I didn't read it carefully... I'm sorry. And thank you for the polite review! I pushed commit.